### PR TITLE
Fixes some bug about alien evolving

### PR DESCRIPTION
**Pull request by phil235**
- Adds a visible message when the larva grows into a humanoid (for consistency with other evolutions).
- Removes some duplicated code in alien evolve code (larva-to-humanoid, praetorian and queen now all use the same proc).
- Alien drones can no longer evolve into a praetorian while inside a pipe (praetorian can't ventcrawl) or other containers.
- Removed an unneeded check in larva_evolve/fire().
- Fixes bug where a larva could evolve despite being incapacitated/dead during the selection of the alien caste.
- Remove outdated code in monkey/attack_hand about disarm and grab intent.
- Evolving into a royal alien no longer deletes what you hold in your hands and the content of your stomach.

### CODE REVIEW

File: code/modules/mob/living/carbon/alien/humanoid/caste/praetorian.dm [hunk 1]
```diff
@@ -54,11 +54,9 @@
 
 /obj/effect/proc_holder/alien/royal/praetorian/evolve/fire(mob/living/carbon/alien/user)
 	if(!alien_type_present(/mob/living/carbon/alien/humanoid/royal/queen))
-		user << "<span class='noticealien'>You begin to evolve!</span>"
-		user.visible_message("<span class='alertalien'>[user] begins to twist and contort!</span>")
+		user.alien_evolve(/mob/living/carbon/alien/humanoid/royal/queen)
 		var/mob/living/carbon/alien/humanoid/royal/queen/new_xeno = new (user.loc)
-		user.mind.transfer_to(new_xeno)
-		qdel(user)
+		user.alien_evolve(new_xeno)
```
Comment (tkdrg) [inline code/modules/mob/living/carbon/alien/humanoid/caste/praetorian.dm]: Did you really mean to alien_evolve twice here?

Comment (phil235) [inline code/modules/mob/living/carbon/alien/humanoid/caste/praetorian.dm]: huh? I only see one alien_evolve() .

File: code/modules/mob/living/carbon/alien/humanoid/caste/praetorian.dm [hunk 2]
```diff
@@ -54,11 +54,9 @@
 
 /obj/effect/proc_holder/alien/royal/praetorian/evolve/fire(mob/living/carbon/alien/user)
 	if(!alien_type_present(/mob/living/carbon/alien/humanoid/royal/queen))
-		user << "<span class='noticealien'>You begin to evolve!</span>"
-		user.visible_message("<span class='alertalien'>[user] begins to twist and contort!</span>")
+		user.alien_evolve(/mob/living/carbon/alien/humanoid/royal/queen)
```
Comment (tkdrg) [inline code/modules/mob/living/carbon/alien/humanoid/caste/praetorian.dm]: @phil235 this is the first one

Comment (phil235) [inline code/modules/mob/living/carbon/alien/humanoid/caste/praetorian.dm]: wtf...how did I miss that????

**Pull request merged**